<a href="https://colab.research.google.com/github/natashanuto/CEE2/blob/main/trabalho_parte1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Trabalho Final: Parte 1
##### Nome: Natasha Nuto
##### Matrícula: 232014852

Exercício 1

In [ ]:
import requests
import pandas as pd


data_inicial = "01/09/2023"
data_final = "01/10/2023"
url = f"https://api.bcb.gov.br/dados/serie/bcdata.sgs.11/dados?formato=json&dataInicial={data_inicial}&dataFinal={data_final}"

response = requests.get(url)

if response.status_code == 200:
  taxa = response.json()

  df = pd.DataFrame(taxa)

else:
  print("Erro ao acessar API do Banco Central.")


Exercício 2

In [ ]:
df["SELIC Diária (%)"] = df["valor"]
df["valor"] = pd.to_numeric(df["valor"]) / 100
df["SELIC Anual (%)"] = round(((1 + df["valor"])**252 - 1) * 100, 2)

Exercício 3

In [ ]:
import pandas as pd

df["SELIC Acumulada (%)"] = round((((1 + df["valor"]).cumprod()) - 1) * 100, 6)

          data     valor SELIC Diária (%)  SELIC Anual (%)  \
0   01/09/2023  0.000490         0.049037            13.15   
1   04/09/2023  0.000490         0.049037            13.15   
2   05/09/2023  0.000490         0.049037            13.15   
3   06/09/2023  0.000490         0.049037            13.15   
4   08/09/2023  0.000490         0.049037            13.15   
5   11/09/2023  0.000490         0.049037            13.15   
6   12/09/2023  0.000490         0.049037            13.15   
7   13/09/2023  0.000490         0.049037            13.15   
8   14/09/2023  0.000490         0.049037            13.15   
9   15/09/2023  0.000490         0.049037            13.15   
10  18/09/2023  0.000490         0.049037            13.15   
11  19/09/2023  0.000490         0.049037            13.15   
12  20/09/2023  0.000490         0.049037            13.15   
13  21/09/2023  0.000473         0.047279            12.65   
14  22/09/2023  0.000473         0.047279            12.65   
15  25/0

Exercício 4

| Datas      |   | SELIC Diária (%) | SELIC Anual (%) | SELIC Acumulada (%) |
|------------|---|------------------|-----------------|---------------------|
| 01/09/2023 |   | 0.049037         | 13.15           | 0.049037            |
| 04/09/2023 |   | 0.049037         | 13.15           | 0.098098            |
| 05/09/2023 |   | 0.049037         | 13.15           | 0.147183            |
| 06/09/2023 |   | 0.049037         | 13.15           | 0.196292            |
| 08/09/2023 |   | 0.049037         | 13.15           | 0.245426            |
| 11/09/2023 |   | 0.049037         | 13.15           | 0.294583            |
| 12/09/2023 |   | 0.049037         | 13.15           | 0.343764            |
| 13/09/2023 |   | 0.049037         | 13.15           | 0.392970            |
| 14/09/2023 |   | 0.049037         | 13.15           | 0.442200            |
| 15/09/2023 |   | 0.049037         | 13.15           | 0.491453            |
| 18/09/2023 |   | 0.049037         | 13.15           | 0.540731            |
| 19/09/2023 |   | 0.049037         | 13.15           | 0.590034            |
| 20/09/2023 |   | 0.049037         | 13.15           | 0.639360            |
| 21/09/2023 |   | 0.047279         | 12.65           | 0.686941            |
| 22/09/2023 |   | 0.047279         | 12.65           | 0.734545            |
| 25/09/2023 |   | 0.047279         | 12.65           | 0.782171            |
| 26/09/2023 |   | 0.047279         | 12.65           | 0.829820            |
| 27/09/2023 |   | 0.047279         | 12.65           | 0.877491            |
| 28/09/2023 |   | 0.047279         | 12.65           | 0.925185            |
| 29/09/2023 |   | 0.047279         | 12.65           | 0.972902            |

\begin{array}{|c|c|c|c|}
\hline
\textbf{Datas} & \textbf{SELIC Diária (%)} & \textbf{SELIC Anual (%)} & \textbf{SELIC Acumulada (%)} \\ \hline
01/09/2023 & 0.049037 & 13.15 & 0.049037 \\
04/09/2023 & 0.049037 & 13.15 & 0.098098 \\
05/09/2023 & 0.049037 & 13.15 & 0.147183 \\
06/09/2023 & 0.049037 & 13.15 & 0.196292 \\
08/09/2023 & 0.049037 & 13.15 & 0.245426 \\
11/09/2023 & 0.049037 & 13.15 & 0.294583 \\
12/09/2023 & 0.049037 & 13.15 & 0.343764 \\
13/09/2023 & 0.049037 & 13.15 & 0.392970 \\
14/09/2023 & 0.049037 & 13.15 & 0.442200 \\
15/09/2023 & 0.049037 & 13.15 & 0.491453 \\
18/09/2023 & 0.049037 & 13.15 & 0.540731 \\
19/09/2023 & 0.049037 & 13.15 & 0.590034 \\
20/09/2023 & 0.049037 & 13.15 & 0.639360 \\
21/09/2023 & 0.047279 & 12.65 & 0.686941 \\
22/09/2023 & 0.047279 & 12.65 & 0.734545 \\
25/09/2023 & 0.047279 & 12.65 & 0.782171 \\
26/09/2023 & 0.047279 & 12.65 & 0.829820 \\
27/09/2023 & 0.047279 & 12.65 & 0.877491 \\
28/09/2023 & 0.047279 & 12.65 & 0.925185 \\
29/09/2023 & 0.047279 & 12.65 & 0.972902 \\
\hline
\end{array}

Exercício 5

In [ ]:
import datetime

df["data"] = pd.to_datetime(df["data"], format="%d/%m/%Y")

         data     valor SELIC Diária (%)  SELIC Anual (%)  SELIC Acumulada (%)
0  2023-09-01  0.000490         0.049037            13.15             0.049037
1  2023-09-04  0.000490         0.049037            13.15             0.098098
2  2023-09-05  0.000490         0.049037            13.15             0.147183
3  2023-09-06  0.000490         0.049037            13.15             0.196292
4  2023-09-08  0.000490         0.049037            13.15             0.245426
5  2023-09-11  0.000490         0.049037            13.15             0.294583
6  2023-09-12  0.000490         0.049037            13.15             0.343764
7  2023-09-13  0.000490         0.049037            13.15             0.392970
8  2023-09-14  0.000490         0.049037            13.15             0.442200
9  2023-09-15  0.000490         0.049037            13.15             0.491453
10 2023-09-18  0.000490         0.049037            13.15             0.540731
11 2023-09-19  0.000490         0.049037            

Exercício 6

In [ ]:
maior = df[df["SELIC Anual (%)"] == max(df["SELIC Anual (%)"])]
menor = df[df["SELIC Anual (%)"] == min(df["SELIC Anual (%)"])]


print("O período com a maior taxa de juros anual foi entre " + maior["data"].iloc[0] + " e " + maior["data"].iloc[-1] + ", no qual a taxa foi de " + str(max(df["SELIC Anual (%)"])) + "%.")
print("O período com a menor taxa de juros anual foi entre " + menor["data"].iloc[0] + " e " + menor["data"].iloc[-1] + ", no qual a taxa foi de " + str(min(df["SELIC Anual (%)"])) + "%.")

O período com a maior taxa de juros anual foi entre 01/09/2023 e 20/09/2023, no qual a taxa foi de 13.15%.
O período com a menor taxa de juros anual foi entre 21/09/2023 e 29/09/2023, no qual a taxa foi de 12.65%.


Exercício 7

In [ ]:
valor = 10000
periodo = pd.date_range(start = "01/09/2023", end = "20/09/2023", freq = "D")

investimento = df[df["data"].isin(periodo)].copy()

investimento["Valor"] = valor


for i in range(1, len(investimento)):
  investimento["Valor"] = investimento.loc[i]["Valor"] * (1 + investimento["valor"])
  investimento["Valor"] = round(investimento["Valor"], 2)

print(investimento)

         data    valor SELIC Diária (%)  SELIC Anual (%)  SELIC Acumulada (%)  \
0  2023-09-01  0.00049         0.049037            13.15             0.049037   
1  2023-09-04  0.00049         0.049037            13.15             0.098098   
2  2023-09-05  0.00049         0.049037            13.15             0.147183   
3  2023-09-06  0.00049         0.049037            13.15             0.196292   
4  2023-09-08  0.00049         0.049037            13.15             0.245426   
5  2023-09-11  0.00049         0.049037            13.15             0.294583   
6  2023-09-12  0.00049         0.049037            13.15             0.343764   
7  2023-09-13  0.00049         0.049037            13.15             0.392970   
8  2023-09-14  0.00049         0.049037            13.15             0.442200   
9  2023-09-15  0.00049         0.049037            13.15             0.491453   
10 2023-09-18  0.00049         0.049037            13.15             0.540731   
11 2023-09-19  0.00049      

Exercício 8

In [ ]:
valor2 = 100
periodo2 = pd.date_range(start = "21/09/2023", end "29/09/2023", freq = "D")